internal/dataplane/parser/translators/ingress.go

```diff
@@ -387,3 +393,97 @@ func MaybePrependRegexPrefixForIngressV1Fn(ingress *netv1.Ingress, applyLegacyHe
 		return lo.ToPtr(MaybePrependRegexPrefix(path, regexPrefix, applyLegacyHeuristic))
 	}
 }
+
+type runeType int
+
+const (
+	runeTypeEscape runeType = iota
+	runeTypeMark
+	runeTypeDigit
+	runeTypePlain
+)
+
+func generateRewriteURIConfig(path string) (string, error) {
+	var out []rune
```

Comment:
Instead of a slice of runes, you can use `strings.Builder` and method `WriteRune`